apps/sms/services/js/service_manager.js

```diff
@@ -0,0 +1,239 @@
+/* global App,
+          ConversationClient,
+          ConversationService,
+          MessagingClient,
+          MessagingService,
+          MozMobileConnectionsClient,
+          MozMobileConnectionsShim,
+          MozSettingsShim
+*/
+
+(function(exports) {
+  'use strict';
+
+  const debug = 1 ?
+    (arg1, ...args) => console.log(`[ServiceManager] ${arg1}`, ...args):
+    () => {};
+
+  const priv = {
+    upgradeKey: Symbol('upgradeKey')
+  };
+
+  const UPGRADE_KEY = 'upgrade-key';
```

Comment:
Well, ```updagrade key``` is another name for ```app instance id``` :) All these dances around the fact that we want to have window#1->SharedWorker->window#1#shim communication channel or if say in another way: we want SharedWorker to request shim in the same window that requested SharedWorker :)

I feel like we should ask Bridge to simplify this problem for us somehow - it's becoming crazy :/ I tried once, but didn't push strong enough :)